app/scripts/background.js

```diff
@@ -327,6 +327,27 @@ function maybeDetectPhishing(theController) {
   );
 }
 
+/**
+ * Overrides the Content-Security-Policy Header, acting as a workaround for an MV2 Firefox Bug.
+ */
+function overrideContentSecurityPolicyHeader() {
+  browser.webRequest.onHeadersReceived.addListener(
+    ({ responseHeaders }) => {
+      for (const header of responseHeaders) {
+        if (header.name.toLowerCase() === 'content-security-policy') {
+          header.value = header.value.replace(
+            /(^|;\s*)script-src([^;]*)/u,
```

Comment:
`\s` matches too many characters. it should be limited to `\t\n\f\r `.

Additionally, "script-src" matching should be case-insensitive while maintaining the original directive name's case in the finally output (incase the dapp's own JavaScript depends on the casing)

Reply:
I also worry about the backtracking in this regular expression.